code/modules/clothing/under/miscellaneous.dm

```diff
@@ -651,12 +651,19 @@
 		if(extinguishes_left)
 			if(next_extinguish > world.time)
 				return
-			next_extinguish = world.time + extinguish_cooldown
-			extinguishes_left--
-			H.visible_message("<span class='warning'>[H]'s suit automatically extinguishes [H.p_them()]!</span>","<span class='warning'>Your suit automatically extinguishes you.</span>")
-			H.ExtinguishMob()
-			new /obj/effect/particle_effect/water(get_turf(H))
-	return 0
+			if((!istype(H.w_uniform, /obj/item/clothing/under/plasmaman/clown)))
+				next_extinguish = world.time + extinguish_cooldown
+				extinguishes_left--
+				H.visible_message("<span class='warning'>[H]'s suit automatically extinguishes [H.p_them()]!</span>","<span class='warning'>Your suit automatically extinguishes you.</span>")
+				H.ExtinguishMob()
+				new /obj/effect/particle_effect/water(get_turf(H))
+			else
+				next_extinguish = world.time + extinguish_cooldown
+				extinguishes_left--
+				H.visible_message("<span class='warning'>[H]'s suit spews out a tonne of space lube!</span>","<span class='warning'>Your suit spews out a tonne of space lube!</span>")
+				H.ExtinguishMob()
+				new /obj/effect/particle_effect/foam(loc) //Truely terrifying.

```

Comment:
Try to avoid dupicate code.

```
next_extinguish = world.time + extinguish_cooldown
extinguishes_left--
H.ExtinguishMob()
if((!istype(H.w_uniform, /obj/item/clothing/under/plasmaman/clown)))
	H.visible_message("<span class='warning'>[H]'s suit automatically extinguishes [H.p_them()]!</span>","<span class='warning'>Your suit automatically extinguishes you.</span>")	
	new /obj/effect/particle_effect/water(get_turf(H))
else	
	H.visible_message("<span class='warning'>[H]'s suit spews out a tonne of space lube!</span>","<span class='warning'>Your suit spews out a tonne of space lube!</span>")
	new /obj/effect/particle_effect/foam(loc) //Truely terrifying.

Reply:
I'm not in a position to test this if I change it for the next week and I'm not comfortable having this P.R. up without knowing it works.

Reply:
If someone can test it then I'd love to.